src/redis-helper.ts

```diff
@@ -119,11 +176,16 @@ export async function getTopTracksForArtist(redisClient: redis.RedisClient,
     const topTracksFromRedis: RedisTrack[]     = [];
     let   topTracksFromSpotify: SpotifyTrack[] = [];
 
-    if (!artist.top_track_ids || artist.top_track_ids.length === 0) {
-        // We've never gotten tracks and saved their ids for this artist, need to call spotify for tracks,
-        // save their ids for this artist, and save the tracks themselves. We cache all track IDs per artist
-        // and tracks themselves, but only return the number requested
-        console.log(`No top track ids for spotify artist ${
+    const lastUpdated = await getArtistTopTracksLastUpdated(redisClient, artist.id);
+
+    console.log(`Last updated top tracks: ${lastUpdated}`);
+
+    if (!artist.top_track_ids || artist.top_track_ids.length === 0 || !lastUpdated ||
+        lastUpdated < (new Date(Date.now() - constants.daysInMsBeforeArtistTracksRefresh))) {
+        // We've never gotten tracks and saved their ids for this artist, or haven't done it for a while, so need to
+        // call spotify for tracks, save their ids for this artist, and save the tracks themselves. We cache all track
+        // IDs per artist and tracks themselves, but only return the number requested
+        console.log(`No/out of date top track ids for spotify artist ${
```

Comment:
can you actually stick the lastupdateddate at the end of this log line please? Just add a period after 'track' at the end and tack on `lastUpdated: ${lastUpdated.toISOString()}`.

Reply:
Done